src/frontend/dsl/sc.rs

```diff
@@ -1,6 +1,6 @@
 use std::{hash::Hash, rc::Rc};
 
-use halo2_proofs::arithmetic::Field;
+use crate::field::Field;
```

Comment:
I did a search over the whole project. Looks like we still have halo2_proofs::arithmetic::Field in halo2.rs and fibonacci.rs.

I think we should also replace them?

Reply:
I changed the fibo example to use our Field, but it does not matter. There is no relation with the inverse thing.

halo2 needs to use its trait